### PR TITLE
added new two options namely "titleElement" and "additionalProperties

### PR DESCRIPTION
**Pull request by talatkuyuk**
Hi, @koka831.

I added new two options namely `titleElement` and `additionalProperties` in order to increase the customization as we have coordinated. 

I have written **additional tests** for the new options as well. 

I also exported the interface `CustomContainerOptions` in order to provide better type for the users who suffers type-matching for the new options, which I also needed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,23 +58,71 @@ import remark from "remark";
...
+
+is going to produce the container below:
+
+```html
+<article class="remark-custom-classname warning" title="My Custom Title" data-type="warning">
+  <p>my paragraph</p>
+</article>
+```
+
+**Note :** The `containerTag` is not prefered to be a `span` or similar, if there is an inner title `div` element. This may cause a problem because of a `div` element under a `span` element.
```
Comment (koka831) [inline README.md:124]: :+1:

File: src/index.ts [hunk 2]
```diff
@@ -17,11 +17,21 @@ interface CustomContainerOptions {
...
+  /**
+   * @defaultValue undefined
+   */
+  additionalProperties?: (className: string, title: string) => Record<string, unknown>;
 }
 
 const DEFAULT_SETTINGS: CustomContainerOptions = {
   className: "remark-container",
   containerTag: "div",
+  titleElement: {},
```
Comment (koka831) [inline src/index.ts:33]: [nit] `{ className: [] }` maybe better?

File: src/index.ts [hunk 3]
```diff
@@ -17,11 +17,21 @@ interface CustomContainerOptions {
...
+  additionalProperties: undefined,
```
Comment (koka831) [inline src/index.ts:34]: IMHO we can use `{}` for its default value for convenience.

File: src/index.ts [hunk 4]
```diff
@@ -3,12 +3,12 @@ import { visit } from "unist-util-visit";
 import type { Plugin, Transformer } from "unified";
 import type { Node, Literal, Parent } from "unist";
 import type { Paragraph } from "mdast";
-import { Data } from "vfile";
+import type { Data } from "vfile";
```
Comment (koka831) [inline src/index.ts:6]: :+1:

File: src/__tests__/options2.test.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import { remark } from "remark";
+import type { Processor } from "unified";
+import type { VFileCompatible } from "vfile";
+import gfm from "remark-gfm";
+import remark2rehype from "remark-rehype";
+import stringify from "rehype-stringify";
+
+import plugin, {type CustomContainerOptions} from "..";
```
Comment (koka831) [inline src/__tests__/options2.test.ts:8]: [nit] 

```suggestion
import plugin, { type CustomContainerOptions } from "..";
```

File: src/__tests__/options2.test.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+const compiler: Processor = remark()
+  .use(gfm)
+  .use(plugin, {
+    additionalProperties: (className, title) => {}
+  } as CustomContainerOptions)
```
Comment (koka831) [inline src/__tests__/options2.test.ts:14]: may I ask you to add a test without any options, just confirm `use(plugin)` works well without breaking change?

**Review by koka831 — COMMENTED** [review]
Overall looks good, with some nits.
Thank you for the PR.

File: src/index.ts (hunk 3 above)
Comment (talatkuyuk) [reply src/index.ts:34]: Do you mean the {} for the titleElement or the additionalProperties ?

**Comment by talatkuyuk** [comment]
Could you have a look at my comments on the conversations about **default value of the options**? Let's finalize the issue.

I fixed the lines you mentioned. I added the requested additional test.  After you agree with my comments I will make a second commit for the issue.

**Comment by koka831** [comment]
Hi, sorry for being late.
I'll review it in this weekend.
Thank you for patience:)

File: src/index.ts (hunk 3 above)
Comment (koka831) [reply src/index.ts:34]: I see it makes sense.

**Review by koka831 — APPROVED** [review]
LGTM, thanks.
I'll merge it after reviewing options4.test.

**Comment by talatkuyuk** [comment]
Thank you for your positive review, I've pushed the additional commit.

**Comment by koka831** [comment]
Thank you @talatkuyuk.
 Could you try to relax these warnings? https://github.com/koka831/remark-custom-container/actions/runs/3664272373/jobs/6194702932

**Comment by talatkuyuk** [comment]
I've pushed the third commit to fix the lint errors by making the parameters in the option "additionalProperties" **as optional**, which is more appropriate

**Review by koka831 — APPROVED** [review]
Looks good, I really appreciate your contribution.
I'll merge it and plan next release.

**Pull request merged**